admin/base/views.py

```diff
@@ -74,3 +57,15 @@ def form_valid(self, form):
     @property
     def success_url(self):
         raise NotImplementedError
+
+
+class GuidView(DetailView):
+    def __init__(self):
+        self.guid = None
+        super(GuidView, self).__init__()
+
+    def get(self, request, *args, **kwargs):
+        try:
+            return super(GuidView, self).get(request, *args, **kwargs)
+        except AttributeError:
```

Comment:
Yes, and its because when loading a guid if it comes back as a `None` then serializing it will return an attribute error which is raised by `get_object` which is then caught by `get` which is what actually returns the rendered web page, in this case a 404.